src/securify/securify_entity.rs

```diff
@@ -0,0 +1,182 @@
+use crate::prelude::*;
+
+impl KeysCollector {
+    pub fn securifying<E: IsEntity>(
+        entity: &E,
+        profile: &Profile,
+        matrix: MatrixOfFactorSources,
+        index_assigner: impl DerivationIndexWhenSecurifiedAssigner,
+        interactors: Arc<dyn KeysDerivationInteractors>,
+    ) -> Result<Self> {
+        let network_id = entity.network_id();
+        let entity_kind = E::kind();
+        KeysCollector::new(
+            profile.factor_sources.clone(),
+            matrix
+                .all_factors()
+                .clone()
+                .into_iter()
+                .map(|f| {
+                    (
+                        f.factor_source_id(),
+                        IndexSet::just(DerivationPath::new(
+                            network_id,
+                            entity_kind,
+                            CAP26KeyKind::T9n,
+                            index_assigner.derivation_index_for_factor_source(
+                                NextFreeIndexAssignerRequest {
+                                    key_space: KeySpace::Securified,
+                                    entity_kind: CAP26EntityKind::Account,
+                                    factor_source_id: FactorSourceIDFromHash::fs0(),
+                                    profile,
+                                    network_id: NetworkID::Mainnet,
+                                },
+                            ),
+                        )),
+                    )
+                })
+                .collect::<IndexMap<FactorSourceIDFromHash, IndexSet<DerivationPath>>>(),
+            interactors,
+        )
+    }
+}
+
+async fn securify_using(
+    address: AccountAddress,
+    matrix: MatrixOfFactorSources,
+    profile: &mut Profile,
+    derivation_index_assigner: impl DerivationIndexWhenSecurifiedAssigner,
+    derivation_interactors: Arc<dyn KeysDerivationInteractors>,
+    gateway: Arc<dyn Gateway>,
+) -> Result<SecurifiedEntityControl> {
+    let account = profile.account_by_address(address.clone())?;
+
+    let keys_collector = KeysCollector::securifying(
```

Comment:
this is simplified, would actually try to first load factor instances from `pre_derived_keys_cache`